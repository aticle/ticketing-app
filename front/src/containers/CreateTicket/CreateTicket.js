// @flow
import React, { Component } from 'react';
import { connect, type Dispatch } from 'react-redux';
import { TextField, Button } from '@material-ui/core';

import { type TicketState, Status } from '../../components/Ticket/Ticket';
import { createTicket } from '../../actions/ticketAction';
import type { State } from '../../reducers';

import './CreateTicket.css';

type Props = {
    createTicket: Dispatch,
    tickets: Array<TicketState>
}

class CreateTicket extends Component<Props, TicketState> {
    state = {
        id: 0,
        status: Status.OPEN,
        title: '',
        description: ''
    };

    handleChange = (name: string) => (e: SyntheticEvent<HTMLInputElement>) => {
        const isNumber = e.currentTarget.type === "number";
        const value = e.currentTarget.value;
        this.setState({
            [name]: isNumber ? Number(value) : value
        });
    }

    handleSubmit = (e: Event) => {
        e.preventDefault();

        const ticket: TicketState = Object.assign({}, this.state);
        this.props.createTicket(ticket);

        this.setState({
            id: this.state.id + 1,
            title: '',
            description: ''
        });
    }

    render() {
        const { id, title, description } = this.state;

        return (
            <form
                className="create-ticket-form"
                onSubmit={this.handleSubmit}
            >
                <h2>Create Ticket</h2>
                <TextField
                    label="ID"
                    id="id"
                    type="number"
                    value={id}
                    required
                    onChange={this.handleChange('id')}
                ></TextField>
                <TextField
                    label="Title"
                    id="title"
                    value={title}
                    required
                    onChange={this.handleChange('title')}
                ></TextField>
                <TextField
                    label="Description"
                    id="description"
                    value={description}
                    required
                    onChange={this.handleChange('description')}
                ></TextField>
                <Button
                    id="submit"
                    variant='contained'
                    type="submit"
                >SUBMIT</Button>
            </form>
        );
    };
}
export const mapStateToProps = (state: State) => {
    return {
        tickets: state.tickets
    }
};
export const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
        createTicket: (ticket: TicketState) => dispatch(createTicket(ticket))
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(CreateTicket);
