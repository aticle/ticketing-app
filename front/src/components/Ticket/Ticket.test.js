import React from 'react';
import { shallow } from 'enzyme';
import CreateTicket from './Ticket';


describe('<CreateTicket />', () => {
    it('renders correctly', () => {
        const createTicket = shallow(<CreateTicket />);
        expect(createTicket).toMatchSnapshot();
    });
});