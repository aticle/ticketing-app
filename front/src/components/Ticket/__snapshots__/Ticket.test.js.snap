// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateTicket /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ticket />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          
        </h1>,
        <h3>
          <i>
            Description: 
          </i>
          <br />
          <p>
            
          </p>
        </h3>,
        <p>
          <b>
            Status: 
          </b>
          OPEN
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i>
              Description: 
            </i>,
            <br />,
            <p>
              
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Description: ",
            },
            "ref": null,
            "rendered": "Description: ",
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <b>
              Status: 
            </b>,
            "OPEN",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Status: ",
            },
            "ref": null,
            "rendered": "Status: ",
            "type": "b",
          },
          "OPEN",
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            
          </h1>,
          <h3>
            <i>
              Description: 
            </i>
            <br />
            <p>
              
            </p>
          </h3>,
          <p>
            <b>
              Status: 
            </b>
            OPEN
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i>
                Description: 
              </i>,
              <br />,
              <p>
                
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Description: ",
              },
              "ref": null,
              "rendered": "Description: ",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <b>
                Status: 
              </b>,
              "OPEN",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Status: ",
              },
              "ref": null,
              "rendered": "Status: ",
              "type": "b",
            },
            "OPEN",
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
